internal/fs/inode/dir.go

```diff
@@ -92,9 +92,14 @@ type DirInode interface {
 	// Return the full name of the child and the GCS object it backs up.
 	CreateChildFile(ctx context.Context, name string) (*Core, error)
 
-	// Create an empty local child file with the supplied (relative) name. Local
-	// file means the object is not yet created in GCS.
-	CreateLocalChildFile(name string) (*Core, error)
+	// LocalChildFileCore returns an empty local child file core.
+	LocalChildFileCore(name string) Core
+
+	// InsertFileIntoTypeCache adds the given file-name to type-cache
+	InsertFileIntoTypeCache(name string)
+
+	// EraseFromTypeCache removes the given file-name from type-cache
```

Comment:
nit: Here, name can file or directory can we change the comment file-name -> child name